### PR TITLE
Shipyard Integration: Add Shipyard Tile to Datadog Integrations

### PR DESCRIPTION
**Pull request by nbeck415**
### What does this PR do?

This PR creates the Shipyard tile for the Shipyard Datadog integration. It has relevant dashboard JSON, readme file with edits from Datadog team, and other required assets. 

### Motivation

We are thrilled to have our Datadog integration up and running, and our next step is offering it as an official integration

### Review checklist

- [x] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [x] Feature or bugfix has tests
- [x] Git history is clean
- [x] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

n/a

### CODE REVIEW

**Comment by hestonhoffman** [comment]
I've opened [a ticket](https://datadoghq.atlassian.net/browse/DOCS-5882) for the docs team to take a look.

File: shipyard/README.md
Comment (anawishnoff) [inline shipyard/README.md:5]:
```suggestion
Shipyard enables teams by providing on-demand (ephemeral) environments for every code change (PR). Shipyard works in your existing CI workflow, empowering teams to test before they merge.
```
We generally avoid marketing-style copy in this repository, so removing this sentence

File: shipyard/manifest.json
Comment (anawishnoff) [inline shipyard/manifest.json:18]: No need to add the logo here, it will automatically appear on your tile. Instead, can you add the sample dashboard image here? Feel free to add other images of your platform as well if you think it's valuable for customers.

File: shipyard/manifest.json
Comment (anawishnoff) [inline shipyard/manifest.json:28]: Please also include the data types that you are submitting into Datadog, or querying from Datadog:

`Submitted Data Type::Logs`
`Queried Data Type::Events`

File: shipyard/manifest.json
Comment (anawishnoff) [inline shipyard/manifest.json:53]:
```suggestion
    "name": "Shipyard",
```

**Review by anawishnoff — COMMENTED** [review]
Hi @nbeck415 , this is looking great - just a few more comments from my end. Can we go ahead and close out the previous PR from Shipyard? Thanks!

File: shipyard/README.md
Comment (urseberry) [inline shipyard/README.md:17]:
```suggestion
**Note:** These instructions assume you have a Shipyard account. If you do not, you can get started [for free][2].
```

**Review by anawishnoff — APPROVED** [review]
LGTM other than my one comment!

File: shipyard/README.md
Comment (urseberry) [inline shipyard/README.md:28]:
```suggestion
   1. *(Optional)* Toggle **Enable Datadog logging for all environments** if you would like Shipyard to automatically send logs for **all** of your environments.
```

File: shipyard/README.md
Comment (urseberry) [inline shipyard/README.md:31]:
```suggestion
   2. Toggle **Enable Datadog logging for this environment**.
```

**Review by urseberry — APPROVED** [review]
Found a couple more formatting issues. Approving so I don't hold up the process.

File: shipyard/manifest.json
Comment (nbeck415) [reply shipyard/manifest.json:29]: @anawishnoff thanks for pointing this out, let me check with my team

Comment (anawishnoff) [reply shipyard/manifest.json:29]: @nbeck415 Hey Natalie, what issues are you facing? Happy to get on a call but I may be able to answer them quickly async.

This set of classifier tags is simply to provide metadata about your integration, and will be displayed on the integration tile. You can read more in our [docs here](https://docs.datadoghq.com/developers/integrations/check_references/#classifier-tags), but we ask all partners to include the types of data that they'll be pulling out of Datadog (queried data types), and the types of data that they'll be sending into Datadog as a part of these classifier tags.

Comment (nbeck415) [reply shipyard/manifest.json:29]: @anawishnoff thanks for the reply -- I am looking at the helm yaml for the integration and trying to figure out which data types we are querying and sending. is there a specific section in that yaml that I'll find the info?

Comment (anawishnoff) [reply shipyard/manifest.json:29]: Not sure if that's present in your code files, but rather something that someone on the product side on your team may know about? I'd talk to whoever conceptualized the integration :)

Comment (nbeck415) [reply shipyard/manifest.json:29]: @anawishnoff I asked and it seems we are just sending shipyard logs, not querying any datadog data. I will update this and then we're good to merge?

Comment (anawishnoff) [reply shipyard/manifest.json:28]: Yup, looks good from my end now, thanks! Once our engineering team reviews, we'll be able to merge this.

File: shipyard/metadata.csv
Comment (moreauowen) [inline shipyard/metadata.csv:1]: Please add the CSV headers to this file here, even if no other rows are present.

**Review by moreauowen — CHANGES_REQUESTED** [review]
Hi there! Left one comment about the `metadata.csv` file. This should fix the failing CI job. Thank you!

File: shipyard/metadata.csv
Comment (nbeck415) [reply shipyard/metadata.csv:1]: just fixed this, thanks!

File: .github/CODEOWNERS
Comment (moreauowen) [inline .github/CODEOWNERS:151]: Looks like this PR introduces changes to the CODEOWNERS of other integration folders. I assume this isn't intentional?

**Review by moreauowen — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS
Comment (nbeck415) [reply .github/CODEOWNERS:151]: Thanks for pointing this out -- I do not recall editing this file and assume it is unintentional?

Comment (nbeck415) [reply .github/CODEOWNERS]: Fixed this!

**Review by moreauowen — APPROVED** [review]

File: shipyard/README.md
Comment (steveny91) [inline shipyard/README.md:58]: non blocking nit:
The link here should ideally be updated later to integrations extras in case the mesmith027 fork ever gets deleted. 

```
All images must be checked into the repo under the `shipyard/images` folder. This image path must be in the form: https://raw.githubusercontent.com/DataDog/integrations-extras/master/shipyard/images/<IMAGE_NAME> Image currently is: https://raw.githubusercontent.com/mesmith027/DataDog-integrations-extras/shipyard-integration/shipyard/images/datadog-input.png
```

**Review by steveny91 — APPROVED** [review]

File: shipyard/README.md
Comment (nbeck415) [reply shipyard/README.md]: Good catch, thank you! Just updated this.

**Comment by moreauowen** [comment]
Hi Shipyard team, can you please merge master into this branch once more? That should re-trigger the failing CI check. Thank you!

**Comment by nbeck415** [comment]
Hey, apologies about that -- when I merged master into this branch it automatically requested review from several users.

**Comment by nbeck415** [comment]
Any ideas on how to pass the failing CI tests?

**Comment by nbeck415** [comment]
> Can you rebase on top of most recent master and solve conflicts? That should clear up the errors. As it is, the PR contains changes from a bunch of integrations which is triggering CI for those and causing the failures.
> 
> It looks like the last merge was done incorrectly. Perhaps removing that last commit and using the GitHub UI to update the PR with master might solve the problem, too.

Hey I think this should be solved, apologies for the messy merge commit. Branch is rebased and the most recent commit is reverted

**Review by xgouchet — APPROVED** [review]

**Comment by emarsha94** [comment]
As mentioned via email, closing this PR for now. Please feel free to re-open once the integration is ready!

**Pull request closed without merging**
